api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -69,13 +69,11 @@
      *
      * @param propertyName
      *             The configuration propertyName.
-     * @return The resolved property value as a String-Optional, which will bypass converters.
+     * @return The resolved property value as a String, which will bypass any converters.
      *
-     * @throws IllegalArgumentException
-     *             is thrown if the propertyName maps to an object that is not a
-     *             String.
+     * 
      */
-    Optional<String> getString(String propertyName);
+    String getRawString(String propertyName);
```

Comment:
> it is counter-productive

Why is that? The value may be present or not, which is exactly the semantics of `Optional`. `Optional` was introduced to represent an optional return value; let's use it instead of requiring people to think about whether it can be null or not. `Optional` conveys that clearly.